### PR TITLE
Explain how to customize notification message in form validation

### PR DESCRIPTION
**Pull request by adguernier**
## Problem

The documentation does not explain how to customize the notification message with form server-side validation.

## Solution

Explain how to customize this message in form server-side validation documentation.

### CODE REVIEW

**Review by fzaninotto — APPROVED** [review]

File: docs/Validation.md [hunk 1]
```diff
@@ -370,6 +370,7 @@ Server-side validation is supported out of the box for `pessimistic` mode only.
 ```
 {
     body: {
+        message: 'Error: my custom message in notification popover',
```
Comment (slax57) [inline docs/Validation.md:373]: I'm confused. Shouldn't the message be outside the body? :thinking: 
See https://github.com/marmelab/react-admin/blob/master/packages/ra-core/src/controller/edit/useEditController.ts#L189 for ref

File: docs/Validation.md [hunk 2]
```diff
@@ -383,7 +384,9 @@ Server-side validation is supported out of the box for `pessimistic` mode only.
 
 **Tip**: The returned validation errors might have any validation format we support (simple strings, translation strings or translation objects with a `message` attribute) for each key.
 
-**Tip**: If your data provider leverages React Admin's [`httpClient`](https://marmelab.com/react-admin/DataProviderWriting.html#example-rest-implementation), all error response bodies are wrapped and thrown as `HttpError`. This means your API only needs to return an invalid response with a json body containing the `errors` key.
+Note: you can use the `message` key to customize the message displayed in the notification popover but it is not mandatory. By default the notification message is the translatable key `ra.notification.http_error`.
```
Comment (slax57) [inline docs/Validation.md:387]: I thought this only worked with React 17. Should we really mention this in the docs then?

File: packages/ra-ui-materialui/src/form/SimpleForm.stories.tsx [hunk 3]
```diff
@@ -145,6 +147,36 @@ export const GlobalValidation = () => (
     </Wrapper>
 );
 
+export const CustomNotificationMessage = () => (
```
Comment (slax57) [inline packages/ra-ui-materialui/src/form/SimpleForm.stories.tsx:150]: IMO this story is misplaced. The logic responsible for this feature is actually implemented in `useEditController`, not in `SimpleForm`.

**Review by slax57 — CHANGES_REQUESTED** [review]

**Comment by slax57** [comment]
The solution documented in this PR doesn't work, and what's more it relies on a race condition between the [http_error notification](https://github.com/marmelab/react-admin/blob/master/packages/ra-core/src/controller/edit/useEditController.ts#L186) and the [invalid_form notification](https://github.com/marmelab/react-admin/blob/master/packages/ra-core/src/form/useNotifyIsFormInvalid.ts#L29), so it's not robust.

Solving this problem can only be addressed by introducing a new field in the server's response, and supporting the ability to return `message: false` to disable the http_error notification, so this will be done in the upcoming v5.

**Pull request closed without merging**